### PR TITLE
SAK-31577 allow navigation bar customisation

### PR DESCRIPTION
**Pull request by bgarciaentornos**

### CODE REVIEW

File: portal/portal-render-engine-impl/pack/src/webapp/vm/morpheus/includePageWithNav.vm [hunk 1]
```diff
@@ -52,24 +61,30 @@
...
 
         
         #foreach( $page in $sitePages.pageNavTools )
             #if (${page.current})
-            <a href="${page.pageResetUrl}" title="$rloader.sit_reset" class="Mrphs-hierarchy-item Mrphs-hierarchy--toolName">
-                <span class="Mrphs-breadcrumb--reset-icon icon-sakai fa fa-share" aria-hidden="true"></span>
-                <span class="Mrphs-breadcrumb--icon ${page.menuClass} $!{page.menuClassOverride}"></span>
-                <span>${page.pageTitle}</span>
+            <a #if(${banner.fontColour}) style="color: ${banner.fontColour};" #end href="${page.pageResetUrl}" title="$rloader.sit_reset" class="Mrphs-hierarchy-item Mrphs-hierarchy--toolName">
+                <span #if(${banner.fontColour}) style="color: ${banner.fontColour};" #end class="Mrphs-breadcrumb--reset-icon icon-sakai fa fa-share" aria-hidden="true"></span>
```
Comment (SedueRey) [inline portal/portal-render-engine-impl/pack/src/webapp/vm/morpheus/includePageWithNav.vm]: why wouldn't this be on the css? 

Colour in span on HTML tags? Why?

**Comment by bgarciaentornos** [comment]
It's an Oxford contribution, we adapted it to morpheus. Guess they just want to be able to customise one, X or all sites without having to get down to the code. So admins of each site can easily do it with just a few clicks. The top bar is not modified. See http://blogs.it.ox.ac.uk/adamweblearn/2014/11/customisable-branding-of-sites/ (although it has changed a bit due to the breadcrumbs/hierarchy).

**Comment by SedueRey** [comment]
I see your point, @bgarciaentornos but, finally, we have achieved to separate style from html itself. I know morpheus is a long road goal and it can be improved in so many ways but I can't see benefits, on 2016, of modifying HTML and not the CSS.

Sorry, this PR adds excessive ravel to the velocity instead adding functionality to morpheus.

**Comment by buckett** [comment]
Would a better way of implementing it be to have some code that dynamically generated the CSS when the properties were set and this dynamically generated CSS was included after the standard CSS thereby overriding the styles specified in the skin?

**Comment by bgarciaentornos** [comment]
The strong point on this functionality is to allow site admins to customise their sites easily. Not sure if that can be done in a different way but what I mean is that "dynamically generated CSS" has to use and take the different site properties. Otherwise the functionality doesn't make sense at all, like they say you can always create many skins.

**Comment by juanjmerono** [comment]
Another easy/fast solution to this is add this property in sakai.properties:

```
portal.include.extrahead=<style>.Mrphs-siteHierarchy { background-color: ...your colour... !important;  }</style>
```

**Comment by bgarciaentornos** [comment]
That's far from the objective of this functionality. Properties have to be defined on each site, plus the way Oxford uses them, they can override the site title, add images and more.

I'll let Matthew and Adam discuss about the possibilities, maybe they keep on using it but don't contribute it as the community won't find it useful.

**Comment by jonespm** [comment]
Yeah I think this the start of a a good idea it really should have

**Comment by ern** [comment]
Closing because this implementation is not ideal. Possible ways of doing this would be to have the following components:
- dynamic css
- a ui to manage css changes

**Pull request closed without merging**